### DISCUSSION TITLE
Dataflow Architecture (DF) v.s. Multilayer Offload Architecture (MO)

### DISCUSSION THREAD

**Discussion by btwbtw01** [Xilinx/finn#516] [Q&A]
Hi, 
I want to ask how the FINN compiler determines to use Dataflow Architecture (DF) or Multilayer Offload Architecture? Can I decide which architecture to use? 
If the architecture is MO, how the dataflow can pipeline?
Suppose the CNN is 8 layers, the architecture has 4 layers. the hardware will do layer1~layer4 first, then do layer5~layer8.
![image](https://user-images.githubusercontent.com/85877169/150989769-9bb7381e-e9f7-4d94-852f-f37802147671.png)
Am I right? 
Can MO do image pipeline, or do sequentially?

**Comment by heborras** [comment]
Hi @btwbtw01,
I think this is almost a bit of a historic question and I'm not sure if I can answer it completely. 
Because I do remember FINN having support for MO architectures in the original paper, however since FINN was rebuilt for open-source development it has dropped support for MO architectures and focused completely on data-flow architectures. At least that is how I understand it historically.

**Reply by btwbtw01** [reply]
Hi, @HenniOVP 
Thanks for letting me know it! That means I can't set large SIMD lanes(Q) or PE number to force the compiler to use Multilayer Offload Architecture because we set the large number that Dataflow Architecture can't fit all the layers like VGG16 or Resnet20.

**Reply by heborras** [reply]
Yes, as far as I know there is no way to get the current FINN compiler to build a network as an MO architecture. Setting the SIMD and PE numbers too high will simply make the synthesis fail.
